File: Python-MySQL - Without Docker/app/files/models/reports.py
import datetime
from files import db
# from files.models.users import User


class Report(db.Model):
    __tablename__ = 'reports'
    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
    title = db.Column(db.String(50), nullable=False)
    submit_date = db.Column(db.DateTime,  default=datetime.datetime.utcnow)
    end_date = db.Column(db.DateTime, nullable=True)
    description = db.Column(db.String(500), nullable=False)
    reporting_user_id = db.Column(
        db.Integer, db.ForeignKey("users.id"))
    report_type_id = db.Column(db.Integer, db.ForeignKey("report_type.id"))
    # operating_user_id = db.Column(db.Integer, db.ForeignKey('users.id'))
    # operating_user = db.relationship("User", back_populates="reports")

    def __repr__(self):

        return f"{{\"id\": \"{self.id}\", \"description\": \"{self.description}\", \"submit_date\": \"{self.submit_date}\", \"end_date\": \"{self.end_date}\", \"title\": \"{self.title}\", \"reporting_user_id\": \"{self.reporting_user_id}\", \"reporting_user\": \"{self.reporting_user.name} {self.reporting_user.surname}\", \"report_type_id\": \"{self.report_type_id}\", \"report_type\": \"{self.report_type.description}\"}}"

File: Node-MySQL/app/models/Reports.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Reports extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      // define association here
    }
  }
  Reports.init({
    id: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true
    },
    title: {
      type: DataTypes.STRING(50),
      allowNull: false,
      validate: {
        notNull: { msg: "Title required!" }
      }
    },
    submitDate: {
      type: DataTypes.DATE,
      allowNull: false,
      defaultValue: DataTypes.NOW,
      validate: {
        notNull: { msg: "SubmitDate required!" }
      }
    },
    endDate: {
      type: DataTypes.DATE,
      allowNull: true
    },
    reportTypeId: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: 'ReportType',
        key: 'id'
      },
      validate: {
        notNull: { msg: "ReportTypeId required!" }
      }
    },
    reportStatusId: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: 'ReportStatus',
        key: 'id'
      },
      validate: {
        notNull: { msg: "ReportStatusId required!" }
      }
    },
    reportingUserId: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: 'Users',
        key: 'id'
      },
      validate: {
        notNull: { msg: "ReportingUserId required!" }
      }
    }
  }, {
    sequelize,
    modelName: 'Reports',
    tableName: 'Reports',
    timestamps: false
  });
  Reports.associate = function (models) {
    Reports.belongsTo(models.ReportType, { foreignKey: 'reportTypeId', as: 'ReportType' });
    Reports.belongsTo(models.ReportStatus, { foreignKey: 'reportStatusId', as: 'ReportStatus' });
    Reports.belongsTo(models.Users, { foreignKey: 'reportingUserId', as: 'ReportingUser' });
  };
  return Reports;
};